src/qtui/status_bar.cc

```diff
@@ -92,8 +97,8 @@ void StatusBar::log_handler (audlog::Level level, const char * file, int line,
 {
     tiny_lock (& message_lock);
 
-    /* do not replace a message of same or higher priority */
-    if (level <= current_message_level)
+    /* do not replace a message of higher priority, or non-critical same-level message */
+    if (level <= current_message_level && level < audlog::Warning)
```

Comment:
Not hiding any warning and higher messages in the statusbar.

OTOH it's a choice to show only the first or the last of a series of such messages generated in a time interval, and there's no general rule that says which is better AFAIK. But OTOH if you show only the first the user doesn't even know if other messages are being generated, and even if you cannot read all messages that fact in itself is an indication that somethings is wrong (a fact you'd miss if only the 1st message is shown).

Reply:
I chose to display the first message rather than the last, because subsequent messages are often either giving more detail about the error (useless if you can't see the first line) or listing additional errors which are side effects of the first one.  It's not perfect, and I agree that there's no general rule to handle all cases perfectly.

Given the limited space in the status bar, some messages *will* be hidden at one point or another.  It would be nice to give an indication of this.  Maybe a suffix "(+ 2 more messages)" that you can click on to open the Log Inspector and see the hidden messages.  Or just make the whole status bar clickable when a warning/error message is displayed.